File: src/main/webapp/js/validator.js
// 유효성 검증 관련 정적 메소드
const Validator = function () {}

// required
Validator.isEmpty = function (input) {
  // 빈공백문자인 경우
  if(input != null){
    return input.trim() == "";
  }else{
    return true;
  } 
}

// 숫자 입력 형식 검증
Validator.isNumber = function (input) {
  return /^[0-9]*$/.test(input);
}

// 이메일 형식 검증
Validator.isEmail = function (input) {
  return /^[a-zA-Z0-9+-\_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/.test(input);
}

// 참고
// const Validate = {};
// Validate.isEmail  = function (input) {
//   return /^[a-zA-Z0-9+-\_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/.test(input);
// }

File: src/main/java/com/ezen/mall/web/common/validate/Validator.java
package com.ezen.mall.web.common.validate;

/**
 * 데이터 유효성 검증을 위한 유틸리티 클래스
 */
public class Validator {
    /**
     * 데이터 입력 여부 검증
     * @param value 검증하고자 하는 문자열
     * @return 값이 있을 경우 true, 없을 경우 false
     */
    public static boolean isEmpty(String value){
        if(value == null || value.trim().length() == 0){
            return true;
        }
        return false;
    }

    /**
     * 이메일 형식 검증
     * @param email 검증하고자 하는 이메일 문자열
     * @return 형식에 맞을 경우 true, 아닐 경우 false
     */
    public static boolean isEmail(String email) {
        String emailV = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,6}$";
        return email.matches(emailV);
    }


    public static void main(String[] args) {
        System.out.println(isEmpty(null));
        System.out.println(isEmpty(""));
        System.out.println(isEmpty(" "));
    }


}
